toolchain/lex/string_literal.h

```diff
@@ -10,70 +10,93 @@
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Support/Allocator.h"
 #include "toolchain/diagnostics/diagnostic_emitter.h"
+#include "toolchain/lex/token_info.h"
 
 namespace Carbon::Lex {
 
 class StringLiteral {
  public:
+  // A string literal's kind.
+  enum class Kind : int8_t {
+    // A character literal is still handled through string literal lexing.
+    // Note the consumer must still validate as a single character.
```

Comment:
Is this note correct? It looks like `StringLiteral::ComputeCharValue` does this validation.